geomesa-fs-bundle/geomesa-fs-processors/src/main/scala/org/geomesa/nifi/processors/fs/FileSystemDataStoreService.scala

```diff
@@ -0,0 +1,26 @@
+/***********************************************************************
+ * Copyright (c) 2015-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.geomesa.nifi.processors.fs
+
+import org.apache.nifi.annotation.documentation.{CapabilityDescription, Tags}
+import org.apache.nifi.components.PropertyDescriptor
+import org.geomesa.nifi.datastore.processor.service.AwsDataStoreService
+import org.geomesa.nifi.datastore.processor.utils.PropertyDescriptorUtils
+import org.locationtech.geomesa.fs.data.FileSystemDataStoreFactory
+import org.locationtech.geomesa.fs.data.FileSystemDataStoreFactory.FileSystemDataStoreParams
+
+@Tags(Array("geomesa", "geotools", "geo", "hdfs", "s3"))
+@CapabilityDescription("Service for connecting to GeoMesa FileSystem stores")
+class FileSystemDataStoreService
+    extends AwsDataStoreService[FileSystemDataStoreFactory](FileSystemDataStoreService.Properties, FileSystemDataStoreParams.ConfigsParam)
+// TODO partition scheme...
```

Comment:
What is the TODO here?

Reply:
as a convenience, there is an option to define the partition scheme in the FS processors. Since the PutGeoMesa processors are now generic, the option is not available any more. i was considering doing something here but it doesn't really make sense at the processor level since it's tied to the feature type, so i just decided not to add it. you can still define it in the feature type configuration or spec string, it's just a bit more inconvenient to re-use a feature type for the fsds... i will remove this TODO.